packaging/metadata/__init__.py

```diff
@@ -0,0 +1,123 @@
+from email.parser import HeaderParser
+from email.message import Message
+from typing import Dict, Any, Iterator
+import json
+from .constants import VERSIONED_METADATA_FIELDS
+
+
+
+def json_form(val: str) -> str:
+    return val.lower().replace("-", "_")
+
+
+class Metadata:
+    def __init__(self, **kwargs: Dict[str, Any]) -> None:
```

Comment:
```suggestion
    def __init__(self, **kwargs: Dict[str, Union[list, str]]) -> None:
```
Use a type union here instead (this isn't 100% right)